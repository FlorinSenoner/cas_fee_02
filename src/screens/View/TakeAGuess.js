import React from 'react'
import PropTypes from 'prop-types'
import Button from 'material-ui/Button'
import TextField from 'material-ui/TextField'
import Dialog, { DialogActions, DialogContent, DialogContentText, DialogTitle } from 'material-ui/Dialog'

export default class TakeAGuess extends React.PureComponent {
  static propTypes = {
    handleGuess: PropTypes.func.isRequired,
  }

  state = {
    open: false,
  }

  handleClickOpen = () => {
    this.setState({ open: true })
  }

  handleClose = () => {
    this.setState({ open: false })
  }

  handleSubmit = () => {
    this.props.handleGuess(document.getElementById('guess').value)
    this.handleClose()
  }

  render() {
    return (
      <div>
        <Button color="primary" onClick={this.handleClickOpen}>
          Make a Guess
        </Button>
        <Dialog open={this.state.open} onClose={this.handleClose} aria-labelledby="form-dialog-title">
          <DialogTitle id="form-dialog-title">Your guess</DialogTitle>
          <DialogContent>
            <DialogContentText>
              Please enter your guess here. Beware! You cannot change it afterwards.
            </DialogContentText>
            <TextField autoFocus margin="dense" id="guess" label="Your guess" fullWidth />
          </DialogContent>
          <DialogActions>
            <Button onClick={this.handleClose} color="primary">
              Cancel
            </Button>
            <Button onClick={this.handleSubmit} color="primary">
              Guess
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    )
  }
}
