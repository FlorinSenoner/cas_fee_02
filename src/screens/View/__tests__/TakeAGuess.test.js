import React from 'react'
import { mount } from 'enzyme'

import TakeAGuess from '../TakeAGuess'
const renderedComponent = mount(<TakeAGuess handleGuess={() => {}} />)

describe('<TakeAGuess />', () => {
  it('renders a TakeAGuess', () => {
    expect(renderedComponent).toMatchSnapshot()
  })
})
