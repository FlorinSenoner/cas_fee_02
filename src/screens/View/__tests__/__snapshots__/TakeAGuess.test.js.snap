// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TakeAGuess /> renders a TakeAGuess 1`] = `
<WithStyles(TakeAGuess)
  handleGuess={[Function]}
>
  <TakeAGuess
    classes={
      Object {
        "button": "TakeAGuess-button-1",
        "rightIcon": "TakeAGuess-rightIcon-2",
      }
    }
    handleGuess={[Function]}
  >
    <WithStyles(Button)
      className="TakeAGuess-button-1"
      color="primary"
      onClick={[Function]}
      variant="raised"
    >
      <Button
        className="TakeAGuess-button-1"
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit-7",
            "disabled": "MuiButton-disabled-12",
            "fab": "MuiButton-fab-13",
            "flatPrimary": "MuiButton-flatPrimary-5",
            "flatSecondary": "MuiButton-flatSecondary-6",
            "fullWidth": "MuiButton-fullWidth-17",
            "keyboardFocused": "MuiButton-keyboardFocused-9",
            "label": "MuiButton-label-4",
            "mini": "MuiButton-mini-14",
            "raised": "MuiButton-raised-8",
            "raisedPrimary": "MuiButton-raisedPrimary-10",
            "raisedSecondary": "MuiButton-raisedSecondary-11",
            "root": "MuiButton-root-3",
            "sizeLarge": "MuiButton-sizeLarge-16",
            "sizeSmall": "MuiButton-sizeSmall-15",
          }
        }
        color="primary"
        disableFocusRipple={false}
        disabled={false}
        fullWidth={false}
        mini={false}
        onClick={[Function]}
        size="medium"
        type="button"
        variant="raised"
      >
        <WithStyles(ButtonBase)
          className="MuiButton-root-3 MuiButton-raised-8 MuiButton-raisedPrimary-10 TakeAGuess-button-1"
          classes={
            Object {
              "keyboardFocused": "MuiButton-keyboardFocused-9",
            }
          }
          disabled={false}
          focusRipple={true}
          onClick={[Function]}
          type="button"
        >
          <ButtonBase
            centerRipple={false}
            className="MuiButton-root-3 MuiButton-raised-8 MuiButton-raisedPrimary-10 TakeAGuess-button-1"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-19",
                "keyboardFocused": "MuiButtonBase-keyboardFocused-20 MuiButton-keyboardFocused-9",
                "root": "MuiButtonBase-root-18",
              }
            }
            disableRipple={false}
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <button
              className="MuiButtonBase-root-18 MuiButton-root-3 MuiButton-raised-8 MuiButton-raisedPrimary-10 TakeAGuess-button-1"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-4"
              >
                Participate 
                <pure(SmsFailed)
                  className="TakeAGuess-rightIcon-2"
                >
                  <SmsFailed
                    className="TakeAGuess-rightIcon-2"
                  >
                    <WithStyles(SvgIcon)
                      className="TakeAGuess-rightIcon-2"
                    >
                      <SvgIcon
                        className="TakeAGuess-rightIcon-2"
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction-24",
                            "colorDisabled": "MuiSvgIcon-colorDisabled-25",
                            "colorError": "MuiSvgIcon-colorError-26",
                            "colorPrimary": "MuiSvgIcon-colorPrimary-22",
                            "colorSecondary": "MuiSvgIcon-colorSecondary-23",
                            "root": "MuiSvgIcon-root-21",
                          }
                        }
                        color="inherit"
                        viewBox="0 0 24 24"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root-21 TakeAGuess-rightIcon-2"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 12h-2v-2h2v2zm0-4h-2V6h2v4z"
                          />
                        </svg>
                      </SvgIcon>
                    </WithStyles(SvgIcon)>
                  </SmsFailed>
                </pure(SmsFailed)>
              </span>
              <WithStyles(TouchRipple)
                center={false}
                innerRef={[Function]}
              >
                <TouchRipple
                  center={false}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child-31",
                      "childLeaving": "MuiTouchRipple-childLeaving-32",
                      "childPulsate": "MuiTouchRipple-childPulsate-33",
                      "ripple": "MuiTouchRipple-ripple-28",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate-30",
                      "rippleVisible": "MuiTouchRipple-rippleVisible-29",
                      "root": "MuiTouchRipple-root-27",
                    }
                  }
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    className="MuiTouchRipple-root-27"
                    component="span"
                    enter={true}
                    exit={true}
                  >
                    <span
                      className="MuiTouchRipple-root-27"
                    />
                  </TransitionGroup>
                </TouchRipple>
              </WithStyles(TouchRipple)>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
    <WithStyles(Dialog)
      aria-labelledby="form-dialog-title"
      onClose={[Function]}
      open={false}
    >
      <Dialog
        aria-labelledby="form-dialog-title"
        classes={
          Object {
            "paper": "MuiDialog-paper-35",
            "paperFullScreen": "MuiDialog-paperFullScreen-40",
            "paperFullWidth": "MuiDialog-paperFullWidth-39",
            "paperWidthMd": "MuiDialog-paperWidthMd-38",
            "paperWidthSm": "MuiDialog-paperWidthSm-37",
            "paperWidthXs": "MuiDialog-paperWidthXs-36",
            "root": "MuiDialog-root-34",
          }
        }
        disableBackdropClick={false}
        disableEscapeKeyDown={false}
        fullScreen={false}
        fullWidth={false}
        maxWidth="sm"
        onClose={[Function]}
        open={false}
        transition={[Function]}
        transitionDuration={
          Object {
            "enter": 225,
            "exit": 195,
          }
        }
      >
        <WithStyles(Modal)
          BackdropProps={
            Object {
              "transitionDuration": Object {
                "enter": 225,
                "exit": 195,
              },
            }
          }
          aria-labelledby="form-dialog-title"
          className="MuiDialog-root-34"
          disableBackdropClick={false}
          disableEscapeKeyDown={false}
          onClose={[Function]}
          open={false}
          role="dialog"
        >
          <Modal
            BackdropComponent={[Function]}
            BackdropProps={
              Object {
                "transitionDuration": Object {
                  "enter": 225,
                  "exit": 195,
                },
              }
            }
            aria-labelledby="form-dialog-title"
            className="MuiDialog-root-34"
            classes={
              Object {
                "hidden": "MuiModal-hidden-42",
                "root": "MuiModal-root-41",
              }
            }
            disableAutoFocus={false}
            disableBackdropClick={false}
            disableEnforceFocus={false}
            disableEscapeKeyDown={false}
            disableRestoreFocus={false}
            hideBackdrop={false}
            keepMounted={false}
            manager={
              ModalManager {
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "modals": Array [],
              }
            }
            onClose={[Function]}
            open={false}
            role="dialog"
          />
        </WithStyles(Modal)>
      </Dialog>
    </WithStyles(Dialog)>
  </TakeAGuess>
</WithStyles(TakeAGuess)>
`;
